pkg/kapp/cmd/appchange/list.go

```diff
@@ -75,14 +110,18 @@ func (t AppChangesTable) Print(ui ui.UI) {
 			uitable.NewHeader("Description"),
 			nsHeader,
 		},
-
 		SortBy: []uitable.ColumnSort{
 			{Column: 1, Asc: false},
 			{Column: 0, Asc: true}, // in case start time are same
 		},
 	}
 
 	for _, change := range t.Changes {
+		if (!t.TimeFlags.BeforeTime.IsZero() && !change.Meta().StartedAt.Before(t.TimeFlags.BeforeTime)) ||
+			(!t.TimeFlags.AfterTime.IsZero() && !change.Meta().StartedAt.After(t.TimeFlags.AfterTime)) {
```

Comment:
```suggestion
		if !(!t.TimeFlags.BeforeTime.IsZero() && change.Meta().StartedAt.Before(t.TimeFlags.BeforeTime) && change.Meta().StartedAt.After(t.TimeFlags.AfterTime))  {
```
How about this? 
Basically, here the internal conditional evaluates to true whenever the current change is in between before time and after time and then the `!` outside will help us continue for all the other changes.